File: NirvanaSMS.Api.CSharp/Core/Utility/ManyToManyMessageBuilder.cs
using NirvanaSMS.Api.Core.Base;
using NirvanaSMS.Api.Core.Entity;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace NirvanaSMS.Api.Core.Utility
{
    public class ManyToManyMessageBuilder : MessageBuilder<ManyToManyMessage, ManyToManyMessageBuilder>
    {
        public List<PhoneAndMessagePair> PhonesAndMessages { get; set; }
        
        public override ManyToManyMessage Build()
        {
            return new ManyToManyMessage(this);
        }

        public ManyToManyMessageBuilder SetPhonesAndMessages(List<PhoneAndMessagePair> phonesAndMessages)
        {
            PhonesAndMessages = phonesAndMessages;
            return this;
        }

        public override ManyToManyMessageBuilder SetHeader(string header)
        {
            Header = header;
            return this;
        }

        public override ManyToManyMessageBuilder SetSendDateTime(DateTime dateTime)
        {
            SendDateTime = dateTime;
            return this;
        }

        public override ManyToManyMessageBuilder SetSendDateTime(string dateTime)
        {
            SendDateTime = DateTime.Parse(dateTime);
            return this;
        }

        public override ManyToManyMessageBuilder SetValidity(int validity)
        {
            ValidityTime = validity;
            return this;
        }
    }
}

File: NirvanaSMS.Api.Java/src/main/java/org/nirvanasms/api/core/utility/ManyToManyMessageBuilder.java
package org.nirvanasms.api.core.utility;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.List;

import org.nirvanasms.api.core.base.MessageBuilder;
import org.nirvanasms.api.core.entity.ManyToManyMessage;
import org.nirvanasms.api.core.entity.OneToManyMessage;
import org.nirvanasms.api.core.entity.PhoneAndMessagePair;

public class ManyToManyMessageBuilder extends MessageBuilder<ManyToManyMessage, ManyToManyMessageBuilder> {

	private List<PhoneAndMessagePair> phonesAndMessages;

	public List<PhoneAndMessagePair> getPhonesAndMessages() {
		return phonesAndMessages;
	}

	public void setPhonesAndMessages(List<PhoneAndMessagePair> phonesAndMessages) {
		this.phonesAndMessages = phonesAndMessages;
	}

	@Override
	public ManyToManyMessageBuilder validity(int validity) {
		super.setValidityTime(validity);
		return this;
	}

	@Override
	public ManyToManyMessageBuilder header(String header) {
		super.setHeader(header);
		return this;
	}

	@Override
	public ManyToManyMessageBuilder sendDateTime(String dateString,String datePattern) {
		Date inputDate;
		try {
			inputDate = DateParser.parse(dateString, datePattern);
			String formattedDateString = DateFormatter.format(inputDate);
			Date formattedDate = DateParser.parse(formattedDateString, Constants.providerXmlDateFormat);
			super.setSendDateTime(formattedDate);
		} catch (ParseException e) {
			e.printStackTrace();
		}
		return this;
	}

	@Override
	public ManyToManyMessageBuilder sendDateTime(Date dateTime) {
		super.setSendDateTime(dateTime);
		return this;
	}

	@Override
	public ManyToManyMessage build() {
		return new ManyToManyMessage(this);
	}

}
